lib/ex_unit/lib/ex_unit/diff.ex

```diff
@@ -24,9 +24,13 @@ defmodule ExUnit.Diff do
 
   # Structs
   def script(%name{} = left, %name{} = right) do
-    left = Map.from_struct(left)
-    right = Map.from_struct(right)
-    script_map(left, right, inspect(name))
+    if Inspect.impl_for(left) != Inspect.Any and inspect(left) != inspect(right) do
+      script_string(inspect(left), inspect(right))
```

Comment:
We should reduce the number of times we call `inspect`.

Reply:
yup, I just wasn't sure how to do it elegantly. How about:

```elixir
  def script(%name{} = left, %name{} = right) do
    inspect_left = inspect(left)
    inspect_right = inspect(right)

    if Inspect.impl_for(left) != Inspect.Any and inspect_left != inspect_right do
      script_string(inspect_left), inspect_right)
    else
      left = Map.from_struct(left)
      right = Map.from_struct(right)
      script_map(left, right, inspect(name))
    end
  end
```

or

```elixir
  def script(%name{} = left, %name{} = right) do
    if Inspect.impl_for(left) != Inspect.Any
      inspect_left = inspect(left)
      inspect_right = inspect(right)

      if inspect_left != inspect_right do
        script_string(inspect_left), inspect_right)
      else
        script_struct(left, right)
      end
    else
      script_struct(left, right)
    end
  end

  defp script_struct(%name{} = left, %name{} = right) do
    left = Map.from_struct(left)
    right = Map.from_struct(right)
    script_map(left, right, inspect(name))
  end
```

Reply:
Second looks good to me.